python/S3_BUCKET_ENCRYPTED_KMS/S3_BUCKET_ENCRYPTED_KMS_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import unittest
+from mock import patch, MagicMock
+from rdklib import Evaluation, ComplianceType
+import rdklibtest
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+RESOURCE_TYPE = 'AWS::S3::Bucket'
+
+#############
+# Main Code #
+#############
+
+MODULE = __import__('S3_BUCKET_ENCRYPTED_KMS')
+RULE = MODULE.S3_BUCKET_ENCRYPTED_KMS()
+
+CLIENT_FACTORY = MagicMock()
+
+#example for mocking S3 API calls
+S3_CLIENT_MOCK = MagicMock()
+
+def mock_get_client(client_name, *args, **kwargs):
+    if client_name == 's3':
+        return S3_CLIENT_MOCK
+    raise Exception("Attempting to create an unknown client")
+
+@patch.object(CLIENT_FACTORY, 'build_client', MagicMock(side_effect=mock_get_client))
+class ComplianceTest(unittest.TestCase):
+
+    test1_list_bucket = {"Buckets":[{"Name":"test-bucket-name-1"}]}
+    test1_encryption = {"ServerSideEncryptionConfiguration":{"Rules":[{"ApplyServerSideEncryptionByDefault":{"SSEAlgorithm":"aws:kms", "KMSMasterKeyID":"dummy-key-id"}}]}}
+
+    test2_list_bucket = {"Buckets":[{"Name":"test-bucket-name-2"}]}
+    test2_encryption = {}
+
+    test3_list_bucket = {"Buckets":[]}
```

Comment:
Please list the test scenarios in the same order as scenarios listed in Gherkin for easy readability. I was expecting test case for NOT_APPLICABLE scenario first as in Gherkin.